src/file_analysis/Tag.h

```diff
@@ -6,105 +6,9 @@
 
 #include "zeek/Tag.h"
 
-namespace zeek::plugin
+namespace zeek::file_analysis
 	{
-template <class T> class TaggedComponent;
-template <class T, class C> class ComponentManager;
-	}
 
-namespace zeek
-	{
-
-class EnumVal;
-
-namespace file_analysis
-	{
-
-class Component;
-
-/**
- * Class to identify a file analyzer type.
- *
- * The script-layer analogue is Files::Tag.
- */
-class Tag : public zeek::Tag
-	{
-public:
-	/*
-	 * Copy constructor.
-	 */
-	Tag(const Tag& other) : zeek::Tag(other) { }
-
-	/**
-	 * Default constructor. This initializes the tag with an error value
-	 * that will make \c operator \c bool return false.
-	 */
-	Tag() : zeek::Tag() { }
-
-	/**
-	 * Destructor.
-	 */
-	~Tag() { }
-
-	/**
-	 * Returns false if the tag represents an error value rather than a
-	 * legal analyzer type.
-	 */
-	explicit operator bool() const { return *this != Error; }
-
-	/**
-	 * Assignment operator.
-	 */
-	Tag& operator=(const Tag& other);
-
-	/**
-	 * Compares two tags for equality.
-	 */
-	bool operator==(const Tag& other) const { return zeek::Tag::operator==(other); }
-
-	/**
-	 * Compares two tags for inequality.
-	 */
-	bool operator!=(const Tag& other) const { return zeek::Tag::operator!=(other); }
-
-	/**
-	 * Compares two tags for less-than relationship.
-	 */
-	bool operator<(const Tag& other) const { return zeek::Tag::operator<(other); }
-
-	/**
-	 * Returns the \c Files::Tag enum that corresponds to this tag.
-	 * The returned value does not have its ref-count increased.
-	 *
-	 * @param etype the script-layer enum type associated with the tag.
-	 */
-	const EnumValPtr& AsVal() const;
-
-	static const Tag Error;
-
-protected:
-	friend class plugin::ComponentManager<Tag, Component>;
-	friend class plugin::TaggedComponent<Tag>;
-
-	/**
-	 * Constructor.
-	 *
-	 * @param type The main type. Note that the \a file_analysis::Manager
-	 * manages the value space internally, so noone else should assign
-	 * main types.
-	 *
-	 * @param subtype The sub type, which is left to an analyzer for
-	 * interpretation. By default it's set to zero.
-	 */
-	explicit Tag(type_t type, subtype_t subtype = 0);
-
-	/**
-	 * Constructor.
-	 *
-	 * @param val An enum value of script type \c Files::Tag.
-	 */
-	explicit Tag(EnumValPtr val);
-	};
+using Tag [[deprecated("Remove in v5.1. Use zeek::Tag.")]] = zeek::Tag;
```

Comment:
Just FYI, this is a breaking change.

While this ensures that the type `zeek::file_analysis::Tag` can still resolve it breaks any code which overloaded functions for different `Tag` types, see e.g., [here](https://github.com/zeek/spicy-plugin/blob/2331bb0522ed104685c94036e60fb49ee43d9821/include/plugin.h#L172-L207). Maybe even worse, code which previously explicitly only matched a `file_analysis::Tag` argument can now match _any `Tag`_ argument; this could show up in e.g., visitors or other template code, or in code which uses function overloading for a `Tag` type and non-`Tag` types.

I think the correct approach instead of using an alias would have been to refactor the old type to contain a `zeek::Tag` and provides its interface, but ensure that it cannot become a `zeek::Tag` (e.g., through implicit conversions).

Here and for other `Tag` types as well.